central/metrics/custom/image_vulnerabilities/tracker_test.go

```diff
@@ -0,0 +1,221 @@
+package image_vulnerabilities
+
+import (
+	"context"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	deploymentMockDS "github.com/stackrox/rox/central/deployment/datastore/mocks"
+	"github.com/stackrox/rox/central/metrics/mocks"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/authproviders"
+	"github.com/stackrox/rox/pkg/grpc/authn/basic"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+)
+
+type image storage.Image
+
+func (i *image) withCVE(cve ...*storage.EmbeddedVulnerability) *image {
+	if i.Scan == nil {
+		i.Scan = &storage.ImageScan{
+			OperatingSystem: "os",
+			Components:      []*storage.EmbeddedImageScanComponent{{}}}
+	}
+	is := i.Scan
+	is.Components[0].Vulns = append(is.Components[0].Vulns, cve...)
+	return i
+}
+
+func makeTestImage(id string) *image {
+	return &image{Id: id}
+}
+
+func (i *image) withTags(tags ...string) *image {
+	for _, tag := range tags {
+		i.Names = append(i.Names, &storage.ImageName{
+			Remote:   "remote",
+			Tag:      tag,
+			Registry: "registry",
+		})
+	}
+	return i
+}
+
+func getTestData() ([]*storage.Deployment, map[string][]*storage.Image) {
```

Comment:
Could you please add `*testing.T` as an argument to all functions that are intended as pure test helpers?